StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -90,8 +92,55 @@ extension EmbeddedPaymentElement: EmbeddedPaymentMethodsViewDelegate {
         delegate?.embeddedPaymentElementDidUpdateHeight(embeddedPaymentElement: self)
     }
 
-    func selectionDidUpdate() {
-        delegate?.embeddedPaymentElementDidUpdatePaymentOption(embeddedPaymentElement: self)
+    func selectionWasTapped(didUpdate: Bool) {
+        // Defer notifying delegate until we are about to exit this function to give time to either allow the new payment option to come from
+        // The new `EmbeddedFormViewController` instance if needed
+        defer {
+            if didUpdate {
+                delegate?.embeddedPaymentElementDidUpdatePaymentOption(embeddedPaymentElement: self)
+            }
+        }
+        
+        guard case let .new(paymentMethodType) = embeddedPaymentMethodsView.selection else {
+            // This can occur when selection is being reset to nothing selected, so don't assert.
+            return
+        }
+
+        guard let presentingVC = presentingViewController else {
+            assertionFailure("Presenting view controller not found, set EmbeddedPaymentElement.presentingViewController.")
+            return
+        }
+
+        let formViewController = EmbeddedFormViewController(
+            configuration: configuration,
+            intent: intent,
+            elementsSession: elementsSession,
+            savedPaymentMethods: savedPaymentMethods,
+            paymentMethodType: paymentMethodType,
+            previousPaymentOption: self.formViewController?.previousPaymentOption,
+            analyticsHelper: analyticsHelper,
+            formCache: formCache
+        )
+        formViewController.delegate = self
+
+        // Only show forms that require user input
+        guard formViewController.collectsUserInput else {
+            self.formViewController = nil  // Clear out any previous form view controller to update self._paymentOption
```

Comment:
1. s/self._paymentOption/self.paymentOption ??
2. I don't understand this comment -- I would have expected to see a didSet for `formViewController` to clear out `paymentOption`